core/src/serve_repair.rs

```diff
@@ -506,6 +524,36 @@ impl ServeRepair {
         Ok(())
     }
 
+    fn repair_stake_stats(
+        &self,
+        bytes_requested: &HashMap<Pubkey, /*bytes*/ u64>,
+    ) -> Option<RepairStakeStats> {
+        let root_bank = self.bank_forks.read().unwrap().root_bank();
+        let epoch_schedule = root_bank.epoch_schedule();
+        let epoch = epoch_schedule.get_epoch(root_bank.slot());
```

Comment:
There is already `Bank::epoch`:
https://github.com/solana-labs/solana/blob/4f232250e/runtime/src/bank.rs#L2348-L2350

Reply:
good call, updated